web_widget_numeric_step/static/src/js/numeric_step.js

```diff
@@ -0,0 +1,239 @@
+/* Copyright 2019 GRAP - Quentin DUPONT
+ * Copyright 2020 Tecnativa - Alexandre Díaz
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html) */
+
+odoo.define('web_widget_numeric_step.field', function (require) {
+    "use strict";
+
+    var Registry = require('web.field_registry');
+    var FieldFloat = require('web.basic_fields').FieldFloat;
+
+    var KeyCodes = {
+        ARROW_DOWN: 40,
+        ARROW_UP: 38,
+    };
+
+    var NumericStep = FieldFloat.extend({
+        template: 'web_widget_numeric_step',
+        className: 'o_field_numeric_step o_field_number',
+        events: _.extend({}, FieldFloat.prototype.events, {
+            'mousedown .btn_numeric_step': '_onStepMouseDown',
+            'touchstart .btn_numeric_step': '_onStepMouseDown',
+            'click .btn_numeric_step': '_onStepClick',
+            'wheel .input_numeric_step': '_onWheel',
+            'keydown .input_numeric_step': '_onKeyDown',
+            'change .input_numeric_step': '_onChange',
+        }),
+        supportedFieldTypes: ['float', 'integer'],
+
+        // Values used to mouse down smooth speed feature
+        DEF_CLICK_DELAY: 400,
+        MIN_DELAY: 50,
+        SUBSTRACT_DELAY_STEP: 25,
+
+        /**
+         * Add global events listeners
+         *
+         * @override
+         */
+        start: function () {
+            var self = this;
+            this._click_delay = this.DEF_CLICK_DELAY;
+            this._autoStep = false;
+            return this._super.apply(this, arguments).then(function () {
+                document.addEventListener(
+                    'mouseup', $.proxy(self._onMouseUp, self), false);
```

Comment:
In general, we should use this form to allow complete overriding:

```suggestion
                    'mouseup', $.proxy(self, "_onMouseUp"), false);
```

This way, the attribute is evaluated lazily too.